src/graphql/shared/root/mutation/user-login.ts

```diff
@@ -18,12 +18,16 @@ const UserLoginInput = GT.Input({
   }),
 })
 
-const UserLoginMutation = GT.Field<{
-  input: {
-    phone: PhoneNumber | InputValidationError
-    code: PhoneCode | InputValidationError
+const UserLoginMutation = GT.Field<
+  null,
+  GraphQLPublicContext | GraphQLAdminContext,
```

Comment:
Can't this be parameterised and declared where it gets added to the schema?
Something like:
```
const UserLoginMutation<T:GraphQLPublicContext | GraphQLAdminContext > = GT.Field<
  null,
 T,
  ```
  
  And then when its used
  ```
  UserLoginMutation<GraphQLPublicContext>
  ```
  ?

Reply:
yeah but this code is going to be deleted so not worth the improvement

Reply:
ie: replaced by rest endpoint. 

this would remove the complexity of having different context for the same function: auth code is better off being outside of the rest of the app